doc/whats_new/v1.4.rst

```diff
@@ -152,6 +152,14 @@ Changelog
   works with pandas copy-on-write mode enabled.
   :pr:`28233` by :user:`John Hopfensperger <s-banach>`.
 
+
+:mod:`sklearn.impute`
+.....................
+
+- |Fix|: :class:`impute.SimpleImputer` now raises an error instead of casting `fill_value` to an integer
+  when input_data is an integer but `fill_value` is a float.
+  :pr:`28365` by :user:`Leo Grinsztajn <LeoGrin>`.
```

Comment:
```suggestion
- |Fix|: :class:`impute.SimpleImputer` now raises an error instead of casting
  `fill_value` to an integer when `input_data` is an integer but `fill_value`
  is a float.
  :pr:`28365` by :user:`Leo Grinsztajn <LeoGrin>`.
```